code/modules/mob/dead/new_player/latejoin_menu.dm

```diff
@@ -0,0 +1,157 @@
+#define JOB_CHOICE_YES "Yes"
+#define JOB_CHOICE_REROLL "Reroll"
+#define JOB_CHOICE_CANCEL "Cancel"
+
+GLOBAL_DATUM_INIT(latejoin_menu, /datum/latejoin_menu, new)
+
+/// Makes a list of jobs and pushes them to a DM list selector. Just in case someone did a special kind of fucky-wucky with TGUI.
+/datum/latejoin_menu/proc/fallback_ui(mob/dead/new_player/user)
+	var/list/jobs = list()
+	for(var/datum/job/job in SSjob.joinable_occupations)
+		jobs += job.title
+
+	var/input = tgui_input_list(user, "Pick a job to join as:", "Latejoin Job Selection", jobs)
```

Comment:
This should be a normal BYOND input, considering tgui is very likely broken at this point.

Reply:
Yea, I was pretty unsure which way you'd prefer at the time, switching it now!